File: src/main/java/com/jinloes/claudereviews/services/stream/StreamEvent.java
package com.jinloes.claudereviews.services.stream;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.Optional;
import lombok.AccessLevel;
import lombok.Data;
import lombok.Getter;

@Data
@Getter(AccessLevel.NONE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class StreamEvent {

    private String type;
    private EventMessage message;
    private String result;

    public String type() {
        return type;
    }

    public Optional<EventMessage> message() {
        return Optional.ofNullable(message);
    }

    public Optional<String> result() {
        return Optional.ofNullable(result);
    }
}

File: core/src/jvmMain/kotlin/com/jinloes/claudereviews/services/stream/StreamEvent.kt
package com.jinloes.claudereviews.services.stream

import java.util.Optional
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class StreamEvent(
    var type: String? = null,
    var subtype: String? = null,
    @SerialName("is_error") @get:JvmName("getIsError") var isError: Boolean = false,
    var message: EventMessage? = null,
    var result: String? = null,
) {
    // Record-style component accessors for Java callers using method-call syntax.
    fun type(): String? = type
    fun subtype(): String? = subtype

    /** Returns whether this is an error event. */
    fun isError(): Boolean = isError

    fun message(): Optional<EventMessage> = Optional.ofNullable(message)
    fun result(): Optional<String> = Optional.ofNullable(result)
}
